src/python/pants/backend/java/target_types.py

```diff
@@ -63,8 +62,10 @@ class JunitTestsGeneratorTarget(Target):
         JavaTestsGeneratorSourcesField,
         Dependencies,
     )
-    help = "A suite of Java tests, run with JUnit."
-
+    help = (
+        "Generate a `junit_test` target for each Java test source file in this directory, "
+        "or the files specified in the `sources` field."
```

Comment:
I fear that could be confusing - it's based solely on the `sources` field, which happens to default to the dir.

```suggestion
        "Generate a `junit_test` target for each file in the `sources` field."
```

Also, the "Java test source file" part reads to me like this generator is going to filter out any non-test files from the `sources` field, which is not the case. If you put `HelloWorld.java` in its `sources`, it will gladly create a `junit_test` target for that.

Reply:
hence the phrasing "for each Java test source file …, or the files specified in the `sources` field" -- the default is to look for files, the alternative is to specify files. I can think about how to make this clearer, but at the moment, I'm fairly OK with it as written

Reply:
Maybe

> Generate a `junit_test` target for each file in the `sources` field (defaults to each Java test source file in the directory).

That would address my concerns very well. Again, my concern is that we need to make 100% unambiguous it is solely based on the `sources` field.

Reply:
That's better, thanks

Reply:
Cool, would be great to update `java_sources` to use similar phrasing